File: leetcode/algorithms/array/maximum-product-of-two-elements-in-an-array.java
/*
    leetcode
    category: "Algorithms"
    problem title: "Maximum Product of Two Elements in an Array"
*/

class Solution {
    public int maxProduct(int[] nums) {
        int[] numsCopy = new int[nums.length];
        
        System.arraycopy(nums,0,numsCopy,0,nums.length);
        
        Arrays.sort(numsCopy);
        
        return Math.max((numsCopy[0]-1)*(numsCopy[1]-1),(numsCopy[numsCopy.length-2]-1)*(numsCopy[numsCopy.length-1]-1));
        
    }
}

File: leetcode/algorithms/array/maximum-product-of-two-elements-in-an-array.js
/*
    leetcode
    category: "Algorithms"
    problem title: "Maximum Product of Two Elements in an Array"
*/

/**
 * @param {number[]} nums
 * @return {number}
 */
var maxProduct = function(nums) {
    
    const numsCopy = [...nums].sort((a,b) => a-b);
    
    return Math.max(numsCopy.slice(0,1).reduce((prod,num) => prod*(num-1),1),numsCopy.slice(numsCopy.length-2,numsCopy.length).reduce((prod,num) => prod*(num-1),1));
    
};